v2/cmd/asoctl/cmd/import_azure_resource.go

```diff
@@ -139,6 +136,9 @@ func importAzureResource(
 		}
 	}
 
+	// Make sure all output is written when we're done
```

Comment:
minor: maybe leave a note that this shouldn't be `defer`ed until right before `Import` is called, because the `Import` call is what actually uses/writes to the progress bar and calls to `Wait` will deadlock if a bar was added but never written to?

Though... if that's actually the case, is it also possible to deadlock if `armIDs` is empty? We'd not take the for loop at all but `importreporter.NewBar` would still be created which is what actually triggers the deadlock?

Reply:
Is it possible to test this case somehow?

Reply:
As long as one bar is created, nothing deadlocks. The call to `importer.Import()` will do that, so we should be safe. I'll refactor the code to tidy things up and add some docs.